tests/python/unittest/test_tvmscript_ops.py

```diff
@@ -103,5 +103,46 @@ def test_get_valid_counts_script_func():
     _check_get_valid_counts_with_numpy(f, (1, 2500, 6), 0.0, 0, 1)
 
 
+@tvm.script.tir
+def ops_with_buffer_slice_indices(data: ty.handle, index: ty.handle) -> None:
+    data_buf = tir.match_buffer(data, (1, 5, 6), "float16")
+    index_buf = tir.match_buffer(index, (1,), "int32")
+    copy_buf = tir.alloc_buffer((1, 5), "float16")
+
+    with tir.block([1, 5], "init") as [vi, vj]:
+        copy_buf[vi, vj] = data_buf[vi, vj, index_buf[0]]
```

Comment:
Can you add the following tests:
- a test that nests buffer accesses three deep. i.e. `data_ buf[index_buf[index_buf[0]]]]`
- a test where the inner buffer dtype is not int32 (should fail).
- a test where the inner buffer is a slice. i.e. `data_buf[index_buf[0,1:2]]`

Reply:
Follow the suggestion of @Hzfengsy, I convert the BufferSlice to BufferLoad in node.py

Meanwhile, add the 3 test cases listed above
1. add `func_with_bufferslice_indices` and `func_with_recursive_bufferslice_indices` in test_tvmscript_complete.py
2. add the fail test case error_bufferslice_index_type in test_tvmscript_error_report.py
3. Given that BufferLoad only accepts elementwise access, add the fail case `error_index_with_stop` `error_bufferslice_index_with_stop` in test_tvmscript_error_report.py